docs/source/howto/data.rst

```diff
@@ -249,11 +247,693 @@ The node class has various methods to set these attributes, such as :py:`~aiida.
+
+*   the ``path``:
+    Here, the user specifies the path along which to join tables as a list of dictionaries, where each list item identifies a vertex in your path.
+    You define the vertex class with the ``cls`` key::
+
+        queryhelp = {
+            'path':[
+                {'cls': Data}
+            ]
+        }
+
+    Each entity in the query has to have a unique tag.
+    If the tag is not provided, it is set to the name of the class.
+    However, this will not work if you choose the same class twice in the query.
+    In this case you have to provide the tag using the ``tag`` key::
+
+        queryhelp = {
+            'path':[
+                {
+                    'cls':Node,
+                    'tag':'node_1'
+                },
+                {
+                    'cls':Node,
+                    'tag':'node_2'
+                }
+            ]
+        }
+
+    You also have to detail some information on the vertex edges, in order to connect them correctly.
+    There are several redundant ways this can be done:
+
+    *   You can specify that this node is an input or output of another node preceding the current one in the list.
+        That other node can be specified by an integer or the class or type.
+        The following examples are all valid joining instructions, assuming there is a structure defined at index 2 of the path with tag "struc1"::
+
+            edge_specification = queryhelp['path'][3]
+            edge_specification['with_incoming'] = 2
+            edge_specification['with_incoming'] = StructureData
+            edge_specification['with_incoming'] = 'struc1'
+            edge_specification['with_outgoing']  = 2
+            edge_specification['with_outgoing']  = StructureData
+            edge_specification['with_outgoing']  = 'struc1'
+
+    *   queryhelp_item['direction'] = integer
+
+        If any of the above specs ("with_outgoing", "with_incoming") were not specified, the key "direction" is looked for.
+        Directions are defined as distances in the tree.
+        1 is defined as one step down the tree along a link.
+        This means that 1 joins the node specified in this dictionary to the node specified on list-item before **as an output**.
+        Direction defaults to 1, which is why, if nothing is specified, this node is joined to the previous one as an output by default.
+        A minus sign reverse the direction of the link.
+        The absolute value of the direction defines the table to join to with respect to your own position in the list.
+        An absolute value of 1 joins one table above, a value of 2 to the table defined 2 indices above.
+        The two following queryhelps yield the same  query::
+
+            from aiida.orm import TrajectoryData
+            from aiida_quantumespresso.calculations.pw import PwCalculation
+            from aiida.orm import Dict
+            qh1 = {
+                'path':[
```

Comment:
We should follow the convetion of one space after the `:`:
```suggestion
                'path': [
```
Same goes for all the other occurrences.